dig/default.go

```diff
@@ -20,30 +20,30 @@
 
 package dig
 
-// Default graph is used for all the top-level inject calls
-var defaultGraph = newGraph()
+// DefaultGraph is used for all the top-level inject calls
+var DefaultGraph Graph = newGraph()
```

Comment:
how about something like this:
``` golang
func New(info auth.CreateAuthInfo, middleware ...OutboundMiddleware) *http.Client
func new(info auth.CreateAuthInfo, digGraph dig.Graph, middleware ...OutboundMiddleware) *http.Client
```
We use the defaultGraph in public API, when we test it we use the private construct function to hijack dependencies.

`http.DefaultClient` is a good point, I never thought that before. I guess things like `http.DefaultClient` and  `http. DefaultTransport` probably are not be changed in most use cases. If golang has immutable data structure, I vote them to be immutable.
But `dig.Graph` cannot be immutable unless it works like `context.Context` which returns a new copy every time after mutation. That may not be a terrible idea.